src/datasource.ts

```diff
@@ -347,9 +347,26 @@ function parseJwtExp(token: string): number {
   return JSON.parse(jsonPayload).exp;
 }
 
-function prependCriblOperator(query: string): string {
-  if (query.trim().startsWith('print')) {
-    return query;
+/**
+ * Cribl Search backend requires a fully-formed query including the "cribl" operator.
+ * Here we're applying best effort to auto-prepending the "cribl" operator where it needs to be.
+ * This is far from 100% foolproof, but covers 99% of the happy paths.
+ * @param query the query as the user entered it
+ * @returns the query with "cribl" operator prepended as needed, if we were able to
+ */
+export function prependCriblOperator(query: string): string {
+  // This tries to capture the first "word" of the root query statement, along with any preceding
+  // "set" or "let" statements, and everything that comes after the first word.  NOTE: This doesn't
+  // touch any "let" statements.  We might enhance that in the future, but for now, users need to
+  // put their own "cribl" operator in those stage statements.
+  const firstWordRegex = /^((?:\s*(?:set|let)\s+[^;]+;)*\s*)((\w+|['"*]).*)$/;
```

Comment:
Should we cover commands as well? I think `.show objects(cribl_search_sample)` would prepend the cribl operator here

Reply:
Good point, but I don't think we need to bother in this context.  People really shouldn't be running dot commands in Grafana.

But to your point, that would try to run that wonky query.  Yeaahhhhh, lemme just make it work.